dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/DateValidator.java

```diff
@@ -55,13 +57,22 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
 
     if (Boolean.TRUE.equals(program.getDisplayIncidentDate())
         && Objects.isNull(enrollment.getOccurredAt())) {
-      reporter.addError(enrollment, E1023, enrollment.getOccurredAt());
+      reporter.addError(enrollment, E1023);
+    }
+
+    validateCompletedDateIsSetOnlyForSupportedStatus(reporter, enrollment);
+  }
+
+  private void validateCompletedDateIsSetOnlyForSupportedStatus(
+      Reporter reporter, Enrollment enrollment) {
+    if (enrollment.getCompletedAt() != null && EnrollmentStatus.ACTIVE == enrollment.getStatus()) {
```

Comment:
Maybe unlikely but still what happens if we add another enrollment status? This would then allow setting the `completedAt`. Is that a good behavior? Or would this be caught by a test. You use a different strategy for events (check for allowed status/here check for disallowed status).

Reply:
You are right, I am changing the logic here